### PR TITLE
Egg spawner script 

### PR DESCRIPTION
**Pull request by priyanka-cs**
The method `SpawnEgg()` in `EggSpawn.cs` spawns eggs randomly based on probability 
Currently GoldenEgg spawn chance is set to 30 and Egg spawn chance is set to 100 in a Temporary Game Manager Object

To test this, uncomment lines 64-96, it'll continuously spawn an egg near the origin once the timer ends

### CODE REVIEW

File: Eggstravaganza/Assets/Scripts/EggSpawn.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    public GameObject prefab;
+    [Range(0f, 100)]
+    public float chance = 100;
+    [HideInInspector]
+    public double weight;
+}
+
+public class EggSpawn : MonoBehaviour
+{
+    public List<Egg> eggs;
```
Comment (picrossin) [inline Eggstravaganza/Assets/Scripts/EggSpawn.cs:18]:
```suggestion
    [SerializeField]
    private List<Egg> m_Eggs;
```
for now perhaps? And can expose as a property later if needed?

File: Eggstravaganza/Assets/Scripts/EggSpawn.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        CalculateWeights();
+    }
+
+    public void SpawnEgg()
+    {
+        var currentEggToSpawn = eggs[ChooseEggToSpawn()];
+        
+        // TODO: Maybe modify spawn height? Not sure what value we'd want yet
+        var randomPosition = Random.insideUnitCircle;
```
Comment (picrossin) [inline Eggstravaganza/Assets/Scripts/EggSpawn.cs:34]: maybe we could also add an exposed property for setting the scale of the spawn area? maybe a vec2 that you multiply `randomPosition` by?

File: Eggstravaganza/Assets/Scripts/EggSpawn.cs (hunk 1 above)
Comment (priyanka-cs) [reply Eggstravaganza/Assets/Scripts/EggSpawn.cs]: whoops you're right, done!

**Review by picrossin — COMMENTED** [review]
Lookin good, found a couple things

**Review by picrossin — APPROVED** [review]
:shipit:

**Pull request merged**